website/docs/spec.md

```diff
@@ -162,6 +162,40 @@ dependencies:
     - libbar
 ```
 
+Sometimes you may need to add extra repositories in order to fulfill the
+specified dependencies.
+You can do this by adding these to the `extra_repos` field.
+The `extra_repos` field takes a list of repositoriy configurations with optional
+public key data and optional repo data (e.g. the actual data of a repository).
+Each of the options takes [source configs](#sources-section).
+
+```yaml
+dependencies:
+  build:
+    - some_dep
+  extra_repos:
+    - config:
+      # source for repo config
+      # This is the data that would go into /etc/yum.repos.d or /etc/apt/sources.list.d, as an example.
```

Comment:
can you add an actual key here such as `foo.repo` and/or `foo.list` so that it fits with the schema?

Reply:
I did not add specific examples here because I have real examples/details in the repositories.md.
The intent here was to just show the structure.
Since this is going to be completely different depending on the distro I was thinking this would be better.
Alternatively I was thinking we could remove the example config and just point at the more detailed document.

Reply:
I do like the idea of just pointing at the more detailed document here to alleviate any confusion!